framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -757,23 +757,28 @@ private void addClass(
               @Nullable @BinaryName String classNameKey,
               @Nullable TypeDeclaration<?> javaParserNode) {
             String className;
-            if (classNameKey == null) {
+            if (classNameKey != null) {
+              className = classNameKey;
+            } else {
               TypeElement classElt = TreeUtils.elementFromDeclaration(tree);
-              className = ElementUtils.getBinaryName(classElt);
-
-              for (TypeElement supertypeElement : ElementUtils.getSuperTypes(classElt, elements)) {
-                String supertypeName = ElementUtils.getBinaryName(supertypeElement);
-                @SuppressWarnings({"signature:assignment", "signature:return"}) // #979?
-                Set<String> supertypeSet =
-                    supertypesMap.computeIfAbsent(className, k -> new TreeSet<>());
-                supertypeSet.add(supertypeName);
-                @SuppressWarnings({"signature:assignment", "signature:return"}) // #979?
-                Set<String> subtypeSet =
-                    subtypesMap.computeIfAbsent(supertypeName, k -> new TreeSet<>());
-                subtypeSet.add(className);
+              if (classElt == null) {
+                className = null;
```

Comment:
How can `classElt` be null here? The previous code seems to:
1. assume that `classElt` is non-null, and
2. rely on the fact that this implies that `className` will never be null: `getBinaryName` will crash with an NPE if its argument is null (it's not null safe)

I'm okay with the change to improve readability, but I'd prefer not to add the null check unless we have an example of code that will cause an NPE here.

Reply:
Narges had this in her changes, and I was perplexed by them.  I'll back this out.